File: android/app/src/main/java/com/smartexpense/android/data/remote/api/NotificationApi.java
package com.smartexpense.android.data.remote.api;

import com.smartexpense.android.data.remote.dto.response.NotificationResponseDto;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.PATCH;
import retrofit2.http.Path;

public interface NotificationApi {
    @GET("api/notifications")
    Call<List<NotificationResponseDto>> getNotifications();

    @PATCH("api/notifications/{id}/read")
    Call<Void> markAsRead(@Path("id") String id);

    @PATCH("api/notifications/read-all")
    Call<Void> markAllAsRead();
}

File: android/app/src/main/kotlin/com/smartexpense/android/data/remote/api/NotificationApi.kt
package com.smartexpense.android.data.remote.api

import com.smartexpense.android.data.remote.dto.response.NotificationResponseDto
import retrofit2.http.GET
import retrofit2.http.PATCH
import retrofit2.http.Path

interface NotificationApi {
    @GET("api/notifications")
    suspend fun getNotifications(): List<NotificationResponseDto>

    @PATCH("api/notifications/{id}/read")
    suspend fun markAsRead(@Path("id") id: String)

    @PATCH("api/notifications/read-all")
    suspend fun markAllAsRead()
}
